src/Spago/Command/Fetch.purs

```diff
@@ -33,18 +34,28 @@ import Registry.Sha256 as Sha256
 import Registry.Solver as Registry.Solver
 import Registry.Version as Registry.Version
 import Registry.Version as Version
-import Spago.Command.Repl as Repl
-import Spago.Config (Dependencies(..), GitPackage, LockfileSettings(..), Package(..), PackageMap, PackageSet(..), Workspace, WorkspacePackage)
+import Spago.Config (Dependencies(..), GitPackage, LockfileSettings(..), Package(..), PackageConfig, PackageMap, PackageSet(..), Workspace, WorkspacePackage)
 import Spago.Config as Config
+import Spago.Core.Config as Core
 import Spago.Db as Db
 import Spago.FS as FS
 import Spago.Git as Git
 import Spago.Lock (LockEntry(..))
 import Spago.Lock as Lock
 import Spago.Paths as Paths
 import Spago.Purs as Purs
+import Spago.Repl as Repl
 import Spago.Tar as Tar
 
+data PackageSelection
+  = SinglePackage
+      { workspacePackage :: WorkspacePackage
+      , configPath :: FilePath
+      , yamlDoc :: YamlDoc Core.Config
+      , transitiveDeps :: PackageMap
+      }
+  | AllWorkspacePackages (Map PackageName PackageMap)
+
```

Comment:
There's a duality between this type above and this type defined in the `Build` module:

```
data SelectedPackageGlob
  = SinglePackageGlobs WorkspacePackage
  | AllWorkspaceGlobs (Array WorkspacePackage)
```

I wonder if it's worth collapsing everything into this one so we avoid the wrapping/unwrapping, something like
```suggestion
data PackageSelection
  = SinglePackage
      { workspacePackage :: WorkspacePackage
      , configPath :: FilePath
      , yamlDoc :: YamlDoc Core.Config
      , transitiveDeps :: PackageMap
      }
  | AllWorkspacePackages (Map WorkspacePackage PackageMap)

```

Reply:
Originally... In `Fetch`, the additional fields (i.e. `configPath` and `yamlDoc`) were added so we wouldn't need to construct them twice / do `case _ of` twice part when running the code under `when ensureRanges` and `when (not $ Array.null packages)`. In `Build`, this extra info isn't needed and may be hard to construct.

So, you might say, let's define a type:
```purs
data PackageSelection a b
  = SinglePackage a
  | AllWorkspacePackges b
```
the works across these two. I guess that works; this just feels super nitpicky to me.

Reply:
> In Build, this extra info isn't needed and may be hard to construct.

Ah you're right indeed - Build needs less stuff. Nevermind then!